internal/ethapi/transaction_args.go

```diff
@@ -258,9 +258,28 @@ func (args *TransactionArgs) ToMessage(globalGasCap uint64, baseFee *big.Int, ru
 
 	// use suggested gasPrice for estimateGas to calculate gasUsed
 	if runMode == core.GasEstimationMode || runMode == core.GasEstimationWithSkipCheckBalanceMode {
-		gasPrice = gasPriceForEstimate.ToInt()
-		gasFeeCap = gasPriceForEstimate.ToInt()
-		gasTipCap = gasPriceForEstimate.ToInt()
+		// use default gasPrice if user does not set gasPrice or gasPrice is 0
+		if args.GasPrice == nil && gasPrice.Cmp(common.Big0) == 0 {
```

Comment:
args.GasPrice.Cmp(common.Big0) == 0 ?

Reply:
In case 2, gasPrice will be 0, so use `default gasPrice`

Reply:
ok